src/Sarif.Multitool/TransformCommand.cs

```diff
@@ -33,21 +45,32 @@ public static int Run(TransformOptions transformOptions)
                     : Formatting.None;
 
                 // Assume the input log is the "other" version
-                if (transformOptions.Version == 2)
+                if (transformOptions.Version == SarifVersion.Current)
                 {
-                    SarifLogVersionOne actualLog = FileHelpers.ReadSarifFile<SarifLogVersionOne>(transformOptions.InputFilePath, SarifContractResolverVersionOne.Instance);
-                    var visitor = new SarifVersionOneToCurrentVisitor();
-                    visitor.VisitSarifLogVersionOne(actualLog);
+                    string inputFilePath = transformOptions.InputFilePath;
+                    string inputVersion = SniffVersion(inputFilePath);
 
-                    FileHelpers.WriteSarifFile(visitor.SarifLog, fileName, formatting);
+                    if (inputFilePath == null || inputFilePath == "1.0.0")
```

Comment:
>inputFilePath [](start = 49, length = 13)

Bug: `inputFilePath` => `inputVersion`

Reply:
I don't understand the null check, either.

---
In reply to: [229513358](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229513358) [](ancestors = 229513358)

Reply:
ouch. mile wide unit test hole, will close. thankfully, i created a nice pattern for doing so. :)

---
In reply to: [229513479](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229513479) [](ancestors = 229513479,229513358)